subprojects/testing-jvm/src/main/java/org/gradle/api/tasks/testing/Test.java

```diff
@@ -770,7 +770,6 @@ public TestFramework testFramework(Closure testFrameworkConfigure) {
      * @return The test framework options.
      */
     @Nested
-    // TODO:LPTR This doesn't resolve any of the nested options for the concrete subtypes
```

Comment:
We should still remove `@Nested` I guess, as neither of the implementations add any `@Input`s etc.

Reply:
That is what I did at first (replacing `@Nested` with `@Internal`), but we actually need to track the options as an input - we track the implementation class as a side effect. This makes it possible to detect changes in the test framework and is actually the only way we currently detect this: https://builds.gradle.org/viewLog.html?buildId=10169674&tab=buildResultsDiv&buildTypeId=Gradle_Check_Quick_Java8_Oracle_Linux_testingJvm

Reply:
Fascinating! Let's add this info as a comment here.

Reply:
Added.